src/SAML2/XML/samlp/LogoutRequest.php

```diff
@@ -278,33 +156,63 @@ public function setSessionIndexes(array $sessionIndexes): void
 
 
     /**
-     * Retrieve the sesion index of the session that should be terminated.
+     * Convert XML into a LogoutRequest
      *
-     * @return string|null The sesion index of the session that should be terminated.
+     * @param \DOMElement $xml The XML element we should load
+     * @return \SAML2\XML\samlp\LogoutRequest
+     * @throws \InvalidArgumentException if the qualified name of the supplied element is wrong
      */
-    public function getSessionIndex(): ?string
+    public static function fromXML(DOMElement $xml): object
     {
-        if (empty($this->sessionIndexes)) {
-            return null;
+        Assert::same($xml->localName, 'LogoutRequest');
+        Assert::same($xml->namespaceURI, LogoutRequest::NS);
+
+        $id = self::getAttribute($xml, 'ID');
+        $version = self::getAttribute($xml, 'Version');
+        $issueInstant = Utils::xsDateTimeToTimestamp(self::getAttribute($xml, 'IssueInstant'));
+        $inResponseTo = self::getAttribute($xml, 'InResponseTo', null);
+        $destination = self::getAttribute($xml, 'Destination', null);
+        $consent = self::getAttribute($xml, 'Consent', null);
+
+        $notOnOrAfter = self::getAttribute($xml, 'NotOnOrAfter', null);
+        if ($notOnOrAfter !== null) {
+            $notOnOrAfter = Utils::xsDateTimeToTimestamp($notOnOrAfter);
         }
 
-        return $this->sessionIndexes[0];
-    }
-
-
-    /**
-     * Set the sesion index of the session that should be terminated.
-     *
-     * @param string|null $sessionIndex The sesion index of the session that should be terminated.
-     * @return void
-     */
-    public function setSessionIndex(string $sessionIndex = null): void
-    {
-        if (is_null($sessionIndex)) {
-            $this->sessionIndexes = [];
-        } else {
-            $this->sessionIndexes = [$sessionIndex];
+        $reason = self::getAttribute($xml, 'Reason', null);
+        $sessionIndexes = Utils::extractStrings($xml, AbstractSamlpElement::NS, 'SessionIndex');
+
+        $issuer = Issuer::getChildrenOfClass($xml);
+        Assert::countBetween($issuer, 0, 1);
+
+        $extensions = Extensions::getChildrenOfClass($xml);
+        Assert::maxCount($extensions, 1, 'Only one saml:Extensions element is allowed.');
+
+        $identifier = self::getIdentifierFromXML($xml);
+        Assert::notNull($identifier, 'Missing <saml:NameID>, <saml:BaseID> or <saml:EncryptedID> in <samlp:LogoutRequest>.');
```

Comment:
Here we should also have a check to verify that whatever we got is either a `BaseID`, a `NameID` or an `EncryptedID` (an `Issuer` element is not allowed here). This is probably something we need to check in other places as well.